dash-pipeline/bmv2/dash_outbound.p4

```diff
@@ -5,212 +5,188 @@
-                             bit<1> meter_policy_en,
-                             bit<16> meter_class) {
-        meta.dst_vnet_id = dst_vnet_id;
-        meta.lkup_dst_ip_addr = overlay_ip;
-        meta.is_lkup_dst_ip_v6 = is_overlay_ip_v4_or_v6;
-        set_route_meter_attrs(meter_policy_en, meter_class);
-    }
-
-    action route_direct(bit<1> meter_policy_en,
-                        bit<16> meter_class) {
-        set_route_meter_attrs(meter_policy_en, meter_class);
-        /* send to underlay router without any encap */
-    }
-
     action drop() {
         meta.dropped = true;
     }
 
-    action route_service_tunnel(bit<1> is_overlay_dip_v4_or_v6,
-                                IPv4ORv6Address overlay_dip,
-                                bit<1> is_overlay_dip_mask_v4_or_v6,
-                                IPv4ORv6Address overlay_dip_mask,
-                                bit<1> is_overlay_sip_v4_or_v6,
-                                IPv4ORv6Address overlay_sip,
-                                bit<1> is_overlay_sip_mask_v4_or_v6,
-                                IPv4ORv6Address overlay_sip_mask,
-                                bit<1> is_underlay_dip_v4_or_v6,
-                                IPv4ORv6Address underlay_dip,
-                                bit<1> is_underlay_sip_v4_or_v6,
-                                IPv4ORv6Address underlay_sip,
-                                dash_encapsulation_t dash_encapsulation,
-                                bit<24> tunnel_key,
-                                bit<1> meter_policy_en,
-                                bit<16> meter_class) {
-        /* Assume the overlay addresses provided are always IPv6 and the original are IPv4 */
-        /* assert(is_overlay_dip_v4_or_v6 == 1 && is_overlay_sip_v4_or_v6 == 1);
-        assert(is_overlay_dip_mask_v4_or_v6 == 1 && is_overlay_sip_mask_v4_or_v6 == 1);
-        assert(is_underlay_dip_v4_or_v6 != 1 && is_underlay_sip_v4_or_v6 != 1); */
-        meta.encap_data.original_overlay_dip = hdr.ipv4.src_addr;
-        meta.encap_data.original_overlay_sip = hdr.ipv4.dst_addr;
-
-        service_tunnel_encode(hdr,
-                              overlay_dip,
-                              overlay_dip_mask,
-                              overlay_sip,
-                              overlay_sip_mask);
-
-        /* encapsulation will be done in apply block based on dash_encapsulation */
-        meta.encap_data.underlay_dip = underlay_dip == 0 ? meta.encap_data.original_overlay_dip : (IPv4Address)underlay_dip;
-        meta.encap_data.underlay_sip = underlay_sip == 0 ? meta.encap_data.original_overlay_sip : (IPv4Address)underlay_sip;
-        meta.encap_data.overlay_dmac = hdr.ethernet.dst_addr;
-        meta.encap_data.dash_encapsulation = dash_encapsulation;
-        meta.encap_data.service_tunnel_key = tunnel_key;
-        set_route_meter_attrs(meter_policy_en, meter_class);
+    action outbound_metadata_publish(MatchStage_t next_stage,
```

Comment:
please group the parameters by routing actions.

Reply:
with comments.

Reply:
Will do